src/UI/Http/Rest/EventSubscriber/ExceptionSubscriber.php

```diff
@@ -78,12 +73,27 @@ private function determineStatusCode(\Exception $exception): int
         // Default status code is always 500
         $statusCode = Response::HTTP_INTERNAL_SERVER_ERROR;
 
-        if ($exception instanceof HttpExceptionInterface) {
-            $statusCode = $exception->getStatusCode();
-        } elseif ($exception instanceof \InvalidArgumentException) {
-            $statusCode = Response::HTTP_BAD_REQUEST;
-        } elseif ($exception instanceof AggregateNotFoundException || $exception instanceof NotFoundException) {
-            $statusCode = Response::HTTP_NOT_FOUND;
+        switch (true) {
```

Comment:
What about using a translation map here instead? `$statusCode = self::EXCEPTION_HTTP_STATUS_CODES[get_class($exception)];`

Reply:
It can hit an undefined index in array and generate a warning in the logs but the switch case not.

Reply:
D'oh.

`$statusCode = self::EXCEPTION_HTTP_STATUS_CODES[get_class($exception)] ?? Response::HTTP_INTERNAL_SERVER_ERROR;`